### PR TITLE
File attachments in the async email sender

### PR DESCRIPTION
**Pull request by zootos**
Adds functionality for file attachments in the asynchronous email sender. Additionally adds a lot of documentation for how the functionality works.

### CODE REVIEW

File: mail/email.py [hunk 1]
```diff
@@ -2,40 +2,114 @@
...
+
+
+def serialize_file(file):
+    """
+    Serializes the content of the file so that it can be sent to the email consumer
+
+    :param file: The file to serialize
+    :return: A tuple of the filename, content and content type
+    """
+    return file.name, file.read(), file.content_type
```
[the remark this answers is not in the document]
Comment (hermabe) [reply mail/email.py]: I have tested it and it works with Djangos [`UploadedFiles`](https://docs.djangoproject.com/en/2.2/ref/files/uploads/#django.core.files.uploadedfile.UploadedFile).  These files are created when a user uploads a file as is the case in the SLA request form. 

We might want to make a helper function to create a Django file from a normal python file object though.

[the remark this answers is not in the document]
Comment (zootos) [reply mail/email.py]: For some reason I forgot about checking normal Python files

**Comment by zootos** [comment]
> I've tested it on the dev-server, but I'm not able to add attachments. I'm not sure how to debug it

Since the file functionality was implemented to allow for use in #245 there is no way to test it without editing the code a bit. The easiest thing is to just check it locally with your own email.

File: mail/email.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (zootos) [reply mail/email.py]: Should work fine now

**Comment by kalkins** [comment]
I've tested it on the dev-server again. The email is sent properly, but the file is not attached. However, when I use the code directly (without `async_to_sync` and `get_channel_layer`) it works fine. Could it be a problem in django, where it can't send attachments async?

This branch is still checkout out on the dev-server if you want to test it yourself.

**Comment by zootos** [comment]
@kalkins I found the reason for this weird behavior. The redis-worker for the dev environment had not been restarted for over 3 weeks, so it used the old code when handling the async request. Restarting it is part of the deploy script of both dev and prod, but I can imagine that you, as I often don't use the deploy script when handling the dev environment. I have restarted it and tested it with async on the dev server (seems to work), but your welcome to test it too.

**Comment by kalkins** [comment]
That makes sense. I'll trust that it works if you say you tested it.

**Pull request merged**